### ISSUE TITLE
Use bundled assets when updating Cordova app 

### ISSUE THREAD

**Issue by martijnwalraven** [meteor/meteor#5639]
It turns out that if there are existing hot code pushed assets, these will always be used instead of the ones bundled with an app update.

One issue with this is that this means there is no way to change the server URL via an update, which is especially important in development. The other is that there may be potential incompatibilities between the native code and the JavaScript code, when removing a Cordova plugin for instance.

The only solution in these cases is to delete and reinstall the app.

**Comment by frapporti** [comment]
We are also looking into the issue, for a quick fix to our production case. Where is the server url stored inside a built iOS app? We looked into NSUserDefaults, NSLibraryDirectory, NSDocumentDirectory so far but no luck.

**Comment by frapporti** [comment]
As a patch, this is working for iPhone.

First, create a function which will be used to delete the resources previously downloaded with hotpush:

```objective-c
-(void) emptyBundles:(int) mainpath withSubFolder:(NSString *)subfolder {
    NSLog(@"EmptyBundles");
    NSFileManager* fm = [[NSFileManager alloc] init];
    NSArray *paths = NSSearchPathForDirectoriesInDomains(mainpath, NSUserDomainMask, YES);
    NSString *documentsDirectory = [NSString stringWithFormat:@"%@/%@", [paths objectAtIndex:0], subfolder];
    
    NSDirectoryEnumerator* en = [fm enumeratorAtPath:documentsDirectory];
    NSError* err = nil;
    BOOL res;
    
    NSString* file;
    while (file = [en nextObject]) {
        NSLog(@"=> %@", file);
        res = [fm removeItemAtPath:[documentsDirectory stringByAppendingPathComponent:file] error:&err];
        if (!res && err) {
            NSLog(@"oops: %@", err);
        }
    }
}
```

Then you call this if the user saved version number is either missing (older version) or different from the previous (user installed update).

```objective-c
NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    if ( ![userDefaults valueForKey:@"version"] || [[NSUserDefaults standardUserDefaults] stringForKey:@"version"] != [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] )
    {
        [self emptyBundles:NSLibraryDirectory withSubFolder:@"NoCloud/meteor/"];
        
        // Setting the current version number to the userdefaults
        [userDefaults setObject:[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] forKey:@"version"];
    }
```

If you want it ready:
[app-delegate.m.txt](https://github.com/meteor/meteor/files/33873/app-delegate.m.txt)
You can just change from .m.txt to .m and replace the app-delegate in your ios project.

This is a drop-in, dirty patch. We kept it in a single file, where the changes are up in the top, so it's easy to replace after every build while an official solution is ready. Also maybe you want to move the version check after `didFinishLaunchingWithOptions`.

**Comment by frapporti** [comment]
For Android, currently our approach has been to edit the meteor_cordova_loader.js file in the www folder, forcing the use of fallback() in loadApp(). This will prevent hotpush from working entirely, but since the play store has no review waiting queue, we decided it was the safest option.

**Comment by martijnwalraven** [comment]
This has been fixed in Meteor 1.3 beta (see https://github.com/meteor/meteor/issues/6050).
